src/test/scala/org/broadinstitute/dsde/workbench/leonardo/auth/sam/SamAuthProviderSpec.scala

```diff
@@ -131,4 +134,66 @@ class SamAuthProviderSpec extends TestKit(ActorSystem("leonardotest")) with Free
     samAuthProvider.samClient.notebookClusters should not contain ((project, name1, userInfo.userEmail) -> Set("connect", "read_policies", "status", "delete", "sync"))
   }
 
+  "should cache hasNotebookClusterPermission results" in isolatedDbTest {
+    val samAuthProvider = getSamAuthProvider
+
+    // cache should be empty
+    samAuthProvider.notebookAuthCache.size shouldBe 0
+
+    // populate backing samClient
+    samAuthProvider.samClient.notebookClusters += (project, name1, userInfo.userEmail) -> Set("sync")
+
+    // call provider method
+    samAuthProvider.hasNotebookClusterPermission(userInfo.userEmail, SyncDataToCluster, project, name1).futureValue shouldBe true
+
+    // cache should contain 1 entry
+    samAuthProvider.notebookAuthCache.size shouldBe 1
+    val key = NotebookAuthCacheKey(userInfo.userEmail, SyncDataToCluster, project, name1, implicitly[ExecutionContext])
+    samAuthProvider.notebookAuthCache.asMap.containsKey(key) shouldBe true
```

Comment:
Why is it necessary to do `asMap` if `samAuthProvider.notebookAuthCache` also has a `get` method?

Reply:
Oh - is it so that we don't cause the cache to refresh potentially?

Reply:
Right - `cache.get(key)` will call the cache loader if the key is not present.

`cache.asMap().get(key)` won't call the cache loader (you are just operating on the backing `Map`)